ambry-rest/src/main/java/com.github.ambry.rest/PublicAccessLogHandler.java

```diff
@@ -181,4 +189,31 @@ private void logError(String msg) {
     logMessage.append(msg);
     publicAccessLogger.logError(logMessage.toString());
   }
+
+  /**
+   * If this is an SSL channel, log information about the peer certificate.
+   * @param ctx the {@link ChannelHandlerContext} for this channel.
+   */
+  private void logSSLInfo(ChannelHandlerContext ctx) {
+    if (sslLogMessage == null) {
+      sslLogMessage = new StringBuilder();
+      sslLogMessage.append("SSL (");
+      try {
+        SslHandler sslHandler = ctx.pipeline().get(SslHandler.class);
+        boolean sslUsed = sslHandler != null;
+        sslLogMessage.append("[used=").append(sslUsed).append("]");
+        if (sslUsed && publicAccessLogger.isCertLoggingEnabled()) {
+          X509Certificate certificate = (X509Certificate) sslHandler.engine().getSession().getPeerCertificates()[0];
+          X500Principal principal = certificate.getSubjectX500Principal();
+          Collection subjectAlternativeNames = certificate.getSubjectAlternativeNames();
+          sslLogMessage.append(", [principal=").append(principal).append("]");
+          sslLogMessage.append(", [san=").append(subjectAlternativeNames).append("]");
```

Comment:
Yeah - doesn't look bad. Let's remove it if you don't have any objections. Would like to not have configs if it isn't necessary.